hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/AssociateTokenRecipientsStep.java

```diff
@@ -168,10 +168,12 @@ private TokenAssociation validateAndBuildAutoAssociation(
         final var tokenRel = tokenRelStore.get(accountId, tokenId);
         final var config = handleContext.configuration();
 
-        if (tokenRel == null && account.maxAutoAssociations() > 0) {
-            validateFalse(
-                    account.usedAutoAssociations() >= account.maxAutoAssociations(),
-                    NO_REMAINING_AUTOMATIC_ASSOCIATIONS);
+        if (tokenRel == null && (account.maxAutoAssociations() > 0 || account.maxAutoAssociations() == -1)) {
+            if (account.maxAutoAssociations() != -1) {
+                validateFalse(
+                        account.usedAutoAssociations() >= account.maxAutoAssociations(),
+                        NO_REMAINING_AUTOMATIC_ASSOCIATIONS);
+            }
```

Comment:
This could be much simpler, and also has a subtle bug if `account.maxAutoAssociations() == 0`:
```suggestion
        if (tokenRel == null) {
            if (account.maxAutoAssociations() != -1) {
                validateFalse(
                        account.usedAutoAssociations() >= account.maxAutoAssociations(),
                        NO_REMAINING_AUTOMATIC_ASSOCIATIONS);
            }
```
The `validateFalse` check handles all the other cases.